conans/model/info.py

```diff
@@ -43,18 +49,25 @@ def dumps(self):
 
     @property
     def sha(self):
-        return "/".join([str(n) for n in [self.name, self.version, self.user, self.channel,
-                                          self.package_id]])
+        vals = [str(n) for n in (self.name, self.version, self.user, self.channel, self.package_id)]
+        # This is done later to NOT affect existing package-IDs (before revisions)
+        return "/".join(vals)
 
     def unrelated_mode(self):
         self.name = self.version = self.user = self.channel = self.package_id = None
 
+    def semver_direct_mode(self):
+        if self._indirect:
```

Comment:
To make sure I understand it well: this conditional is here to allow package_id() defined in the recipe even if the general conf indicates something different, right?

Reply:
The conditional is the implementation of the current default, which we called "semver_direct_mode"

Indeed, the new ``semver_direct_mode()`` methods are for those who want to use that mode even when in conan.conf another mode is defined.